google-cloud-bigquery/lib/google/cloud/bigquery/service.rb

```diff
@@ -590,22 +590,20 @@ def initialize retries: nil, reasons: nil, backoff: nil
           def execute
             current_retries = 0
             loop do
-              begin
-                return yield
-              rescue Google::Apis::Error => e
-                raise e unless retry? e.body, current_retries
-
-                @backoff.call current_retries
-                current_retries += 1
-              end
+              return yield
+            rescue Google::Apis::Error => e
+              raise e unless retry? e.body, current_retries
+
+              @backoff.call current_retries
+              current_retries += 1
             end
           end
 
           protected
 
           def retry? result, current_retries #:nodoc:
-            if current_retries < @retries
-              return true if retry_error_reason? result
+            if current_retries < (@retries) && (retry_error_reason? result)
```

Comment:
This is kind of bizarre paren usage. Why not just:

```ruby
if current_retries < @retries && retry_error_reason?(result)
```

And in fact, I think we can just drop the if and return statements, and just return the expression result as-is.

Reply:
Changed parens as suggested, thanks.